static/js/typing.js

```diff
@@ -51,11 +52,15 @@ function get_current_time() {
 }
 
 function notify_server_start(user_ids_array) {
-    send_typing_notification_ajax(user_ids_array, "start");
+    if (page_params.send_private_typing_notifications) {
+        send_typing_notification_ajax(user_ids_array, "start");
+    }
 }
```

Comment:
I think this is a good place to do this check to keep everything simple. It would also allow us to extend this for `send_stream_typing_notifications` easily.

Reply:
Yeah, I think that's a reasonable interface.

Reply:
Thanks for confirming!